### PR TITLE
Feature/rename tee enclave provider

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

**Comment by jbern0rd** [comment]
I am puzzled by a point.
I don't know where `TeeEnclaveProviderConverter` is being used.
It does not seem to be the case and may be removable.
What do you think about this ?

**Review by mcornaton — APPROVED** [review]
Thanks!

**Comment by mcornaton** [comment]
About https://github.com/iExecBlockchainComputing/iexec-sms/pull/128#issuecomment-1269760002: it is used to convert URIs. Without this class, the following wouldn't work if the path variable is not provided in upper case:
https://github.com/iExecBlockchainComputing/iexec-sms/blob/7c90b33ea091f2907a39250a79b9148ba8e5f305/src/main/java/com/iexec/sms/tee/TeeController.java#L91-L93

E.g., the following wouldn't work without the converter:
- `GET /properties/scone`
- `GET /properties/Scone`
- `GET /properties/Gramine`
- `GET /properties/gramine`
Only the following would work:
- `GET /properties/SCONE`
- `GET /properties/GRAMINE`

That's what's explained in `TeeEnclaveProviderConverter`'s JavaDoc ;)

**Review by mcornaton — APPROVED** [review]
Thanks!

File: iexec-sms-library/src/main/java/com/iexec/sms/api/config/GramineServicesProperties.java [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
 
-    public GramineServicesProperties() {
-        super(TeeEnclaveProvider.GRAMINE);
-    }
-
-    public GramineServicesProperties(TeeAppProperties preComputeProperties,
-                                     TeeAppProperties postComputeProperties) {
-        super(TeeEnclaveProvider.GRAMINE, preComputeProperties, postComputeProperties);
+    @JsonCreator
+    public GramineServicesProperties(@JsonProperty("preComputeProperties") TeeAppProperties preComputeProperties,
```
Comment (jeremyjams) [inline iexec-sms-library/src/main/java/com/iexec/sms/api/config/GramineServicesProperties.java:28]: :pray:

File: iexec-sms-library/src/main/java/com/iexec/sms/api/config/TeeServicesProperties.java [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
-@AllArgsConstructor
 @Getter
-@Setter
+@AllArgsConstructor
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "teeFramework", visible = true)
+@JsonSubTypes({
+        @JsonSubTypes.Type(name = "SCONE", value = SconeServicesProperties.class),
+        @JsonSubTypes.Type(name = "GRAMINE", value = GramineServicesProperties.class)
+})
+// TODO upgrade to sealed class in Java 17
```
Comment (jeremyjams) [inline iexec-sms-library/src/main/java/com/iexec/sms/api/config/TeeServicesProperties.java:32]: :white_check_mark:

**Review by jeremyjams — APPROVED** [review]
Thanks!

**Pull request merged**